install.go

```diff
@@ -287,28 +287,42 @@ func (i *Install) SelectGoodInstallLoc() error {
 var errNoGoodInstall = fmt.Errorf("could not find good install location")
 
 func findGoodInstallDir() (string, error) {
-	// gopath setup?
-	gopath := os.Getenv("GOPATH")
-	if gopath != "" {
-		return filepath.Join(gopath, "bin"), nil
+	// Gather some candidate locations
+	// The first ones have more priority than the last ones
+	var candidates []string
+
+	// First candidate GOBIN
+	gobin := os.Getenv("GOBIN")
```

Comment:
lets not install to $GOBIN, thats not a common thing and I don't think its recommended by anyone